src/libsyntax/std_inject.rs

```diff
@@ -47,6 +47,10 @@ pub fn maybe_inject_crates_ref(
 
     // the first name in this list is the crate name of the crate with the prelude
     let names: &[&str] = if attr::contains_name(&krate.attrs, "no_core") {
+        // Use `crate::module` because `::module` is not allowed on the Rust 2018 edition.
+        if rust_2018 {
+            INJECTED_CRATE_NAME.with(|opt_name| opt_name.set(Some("crate")));
```

Comment:
Isn't the convention that `INJECTED_CRATE_NAME` of `None` corresponds to `#![no_core]`?
That means you can handle that case where `INJECTED_CRATE_NAME` is used, without changing this file.